heron/packing/tests/java/com/twitter/heron/packing/roundrobin/ResourceCompliantRRPackingTest.java

```diff
@@ -0,0 +1,570 @@
+
+    Assert.assertEquals(packingPlanExplicitResourcesConfig.containers.size(), numContainers);
+
+
+    Assert.assertEquals(Math.round(7 * instanceCpuDefault
+            + (10 * 7 * instanceCpuDefault) / 100 + instanceCpuDefault),
+        (long) packingPlanExplicitResourcesConfig.resource.cpu);
+
+    Assert.assertEquals(7 * instanceRamDefault
+            + (10 * 7 * instanceRamDefault) / 100
+            + instanceRamDefault,
+        packingPlanExplicitResourcesConfig.resource.ram);
+
+    Assert.assertEquals(7 * instanceDiskDefault
+            + (10 * 7 * instanceDiskDefault) / 100
+            + instanceDiskDefault,
+        packingPlanExplicitResourcesConfig.resource.disk);
+
+    for (PackingPlan.ContainerPlan containerPlan
+        : packingPlanExplicitResourcesConfig.containers.values()) {
+      Assert.assertEquals(Math.round(7 * instanceCpuDefault
+              + (10 * 7 * instanceCpuDefault) / 100),
+          (long) containerPlan.resource.cpu);
+
+      Assert.assertEquals(7 * instanceRamDefault
+              + (10 * 7 * instanceRamDefault) / 100,
+          containerPlan.resource.ram);
+
+      Assert.assertEquals(7 * instanceDiskDefault
+              + (10 * 7 * instanceDiskDefault) / 100,
+          containerPlan.resource.disk);
+
+      // All instances' resource requirement should be equal
+      // So the size of set should be 1
+      Set<PackingPlan.Resource> resources = new HashSet<>();
+      for (PackingPlan.InstancePlan instancePlan : containerPlan.instances.values()) {
+        resources.add(instancePlan.resource);
+      }
+
+      Assert.assertEquals(1, resources.size());
+      Assert.assertEquals(instanceRamDefault, resources.iterator().next().ram);
+    }
+  }
+
+  /**
+   * Test the scenario where container level resource config are set
+   */
+  @Test
+  public void testContainerRequestedResources2() throws Exception {
+    int numContainers = 2;
+    int spoutParallelism = 4;
+    int boltParallelism = 3;
+
+    // Set up the topology and its config
+    com.twitter.heron.api.Config topologyConfig = new com.twitter.heron.api.Config();
+    topologyConfig.put(com.twitter.heron.api.Config.TOPOLOGY_STMGRS, numContainers);
+    // Explicit set resources for container
+    long containerRam = 10L * Constants.GB;
+    long containerDisk = 20L * Constants.GB;
+    float containerCpu = 30;
+
+    topologyConfig.setContainerRamRequested(containerRam);
+    topologyConfig.setContainerDiskRequested(containerDisk);
+    topologyConfig.setContainerCpuRequested(containerCpu);
+    TopologyAPI.Topology topologyExplicitResourcesConfig =
+        getTopology(spoutParallelism, boltParallelism, topologyConfig);
+    PackingPlan packingPlanExplicitResourcesConfig =
+        getResourceCompliantRRPackingPlan(topologyExplicitResourcesConfig);
+
+    Assert.assertEquals(packingPlanExplicitResourcesConfig.containers.size(), numContainers);
+
+    Assert.assertEquals((long) (Math.round(4 * instanceCpuDefault
```

Comment:
Got it. I thought they were similar except for the constants. The logic behind the math in the assertions is not entirely obvious so if there's something that could be done to help clarify the equation to the reader that would help.